classes/class-twentytwenty-customize.php

```diff
@@ -67,11 +67,71 @@ public static function register( $wp_customize ) {
 				)
 			);
 
+			/**
+			 * Implementation for the accent color.
+			 * This is different to all other color options because of the accessibility enhancements.
+			 * The control is a hue-only colorpicker, and there is a separate setting that holds values
+			 * for other colors calculated based on the selected hue and various background-colors on the page.
+			 *
+			 * @since 1.0.0
+			 */
+
+			// Add the setting for the hue colorpicker.
+			$wp_customize->add_setting(
+				'accent_hue',
+				array(
+					'default'           => 344,
+					'type'              => 'theme_mod',
+					'sanitize_callback' => 'absint',
+					'transport'         => 'postMessage',
+				)
+			);
+
+			// Add setting to hold colors derived from the accent hue.
+			$wp_customize->add_setting(
+				'accent_accessible_colors',
+				array(
+					'default'           => array(), // TODO: We need some sane defaults here.
+					'type'              => 'theme_mod',
+					'transport'         => 'postMessage',
+					'sanitize_callback' => function( $value ) { // TODO: This should probably be a normal mehod in this object instead of a closure.
+						$value = is_array( $value ) ? $value : array();
+						foreach ( $value as $context => $values ) {
+							$value[ $context ] = array(
+								'text' => ( isset( $value[ $context ]['text'] ) ) ? $value[ $context ]['text'] : '#000000',
+								'accent' => ( isset( $value[ $context ]['accent'] ) ) ? $value[ $context ]['accent'] : '#CD2653',
+							);
+						}
+						return $value;
```

Comment:
This should not be an anonymous function, it should instead be a normal method in the class. Some inline docs too wouldn't hurt.

Reply:
Done